File: PHP/DesignPattern/FactoryMethod/Pizza.php
<?php
	/**
	* 产品类
	* 工厂产生产品，对于PizzaStore来说，产品就是Pizza
	*/
	abstract class Pizza {
		public $name;

		public function prepare() {
			echo "Preparing " . $this->name . "<br>";
		}

		public function bake() {
			echo "Baking " . "<br>";
		}

		public function cut() {
			echo "Cutting..." . "<br>";
		}

		public function box() {
			echo "Boxing.." . "<br>";
		}

		public function getName() {
			return $this->name . "<br>";
		}
	}

File: Java/FactoryMethod/src/com/base/Pizza.java
package com.base;

import java.util.ArrayList;

/**
 * Created by hhq on 1/14/15.
 */
public abstract class Pizza {
    String name;
    String dough;
    String sauce;
    ArrayList toppings = new ArrayList();
    
    void prepare() {
        System.out.println("Preparing " + name);
        System.out.println("Tossing dough");
        System.out.println("Adding sauce...");
        System.out.println("Adding toppings:");
        for (int i = 0; i < toppings.size(); i++) {
            System.out.println("  " + toppings.get(i));
        }
    }
    
    void bake() {
        System.out.println("Bake for 25 minutes at 350");
    }
    
    void cut() {
        System.out.println("Cutting the pizza into diagonal slices");
    }
    
    void box() {
        System.out.println("Place pizza in official com.base.PizzaStore box");
    }
    
    public String getName() {
        return name;
    }
}
